bench-vote/src/main.rs

```diff
@@ -0,0 +1,331 @@
+#![allow(clippy::arithmetic_side_effects)]
+
+use {
+    clap::{crate_description, crate_name, Arg, Command},
+    crossbeam_channel::unbounded,
+    solana_client::connection_cache::ConnectionCache,
+    solana_connection_cache::client_connection::ClientConnection,
+    solana_net_utils::bind_to_unspecified,
+    solana_sdk::{
+        hash::Hash, message::Message, signature::Keypair, signer::Signer, transaction::Transaction,
+    },
+    solana_streamer::{
+        packet::PacketBatchRecycler,
+        streamer::{receiver, PacketBatchReceiver, StreamerReceiveStats},
+    },
+    solana_vote_program::{vote_instruction, vote_state::Vote},
+    std::{
+        cmp::max,
+        net::{IpAddr, Ipv4Addr, SocketAddr, UdpSocket},
+        sync::{
+            atomic::{AtomicBool, AtomicUsize, Ordering},
+            Arc,
+        },
+        thread::{self, spawn, JoinHandle, Result},
+        time::{Duration, Instant, SystemTime},
+    },
+};
+
+fn sink(
+    exit: Arc<AtomicBool>,
+    rvs: Arc<AtomicUsize>,
+    r: PacketBatchReceiver,
+    verbose: bool,
+) -> JoinHandle<()> {
+    spawn(move || {
+        let mut last_report = Instant::now();
+        let interval: Duration = Duration::from_secs(5);
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                return;
+            }
+            let timer = Duration::new(1, 0);
+            if let Ok(packet_batch) = r.recv_timeout(timer) {
+                rvs.fetch_add(packet_batch.len(), Ordering::Relaxed);
+            }
+
+            let count = rvs.load(Ordering::Relaxed);
+
+            if verbose &&
+            /*count % 1000 == 0 && count != 0 &&*/
+            last_report.elapsed() > interval
+            {
+                println!("Received txns count: {count}");
+                last_report = Instant::now();
+            }
+        }
+    })
+}
+
+const TRANSACTIONS_PER_THREAD: u64 = 1000000; // Number of transactions per thread
+
+fn main() -> Result<()> {
+    let mut num_sockets = 1usize;
+
+    let matches = Command::new(crate_name!())
+        .about(crate_description!())
+        .version(solana_version::version!())
+        .arg(
+            Arg::new("num-recv-sockets")
+                .long("num-recv-sockets")
+                .value_name("NUM")
+                .takes_value(true)
+                .help("Use NUM receive sockets"),
+        )
+        .arg(
+            Arg::new("num-producers")
+                .long("num-producers")
+                .value_name("NUM")
+                .takes_value(true)
+                .help("Use this many producer threads."),
```

Comment:
This is copied for several of arguments below
```
.help("Use this many producer threads.")
```

Reply:
done